Fix/ Mosque not found on fetch issue

ghassenbenzahra123 opened the pull request:
📝 **Summary**
---

- Only set mosque value to null on failure to fetch when no cached data exists.

**Checklist:**
---
- [x] **Coding Standards:** I have reviewed my code to ensure it follows the project's coding standards.
- [x] **Testing:** I have tested the changes and they work as expected.
- [x] **Merge Conflicts:** I have resolved any merge conflicts with the latest main/development branch.
- [x] **Branch Status:** The branch is up-to-date with the target branch (main/development).

YassinNouh21 commented:
1. When a mosque is successfully fetched:
   ```dart
   _mosqueSubscription = mosqueStream.listen(
     (e) async {
       mosque = e;
       await prefs.setBool(MosqueManagerConstant.hasCachedMosque, true);
       _updateFlashEnabled();
       notifyListeners();
     },
     onError: onItemError,
   );
   ```

2. When an error occurs (e.g., mosque not found):
   ```dart
   onItemError(e, stack) {
     logger.e(e, stackTrace: stack);
     bool hasCachedMosque = prefs.getBool(MosqueManagerConstant.hasCachedMosque) ?? false;
     if (!hasCachedMosque) {
       mosque = null;
       notifyListeners();
     }
     throw e;
   }
   ```

In the scenario you described:

1. The user initially requests a mosque that is found. The `hasCachedMosque` flag is set to `true`.
2. Later, if that mosque becomes not found (e.g., deleted from the server), the `onItemError` function will be called.
3. Because `hasCachedMosque` is `true`, the error handling won't set `mosque` to null or notify listeners.
4. The error is still thrown (`throw e;` at the end of `onItemError`).

This implementation will indeed cause an issue. The error will be thrown, but the UI won't be updated to reflect that the mosque is no longer available. The app will continue to display the cached data of a mosque that no longer exists.

YassinNouh21 commented:
so we are trying to revert to the previous implementation that if the mosque not found it will do nothing

ghassenbenzahra123 commented:
> 1. When a mosque is successfully fetched:
>    ```dart
>    _mosqueSubscription = mosqueStream.listen(
>      (e) async {
>        mosque = e;
>        await prefs.setBool(MosqueManagerConstant.hasCachedMosque, true);
>        _updateFlashEnabled();
>        notifyListeners();
>      },
>      onError: onItemError,
>    );
>    ```
> 2. When an error occurs (e.g., mosque not found):
>    ```dart
>    onItemError(e, stack) {
>      logger.e(e, stackTrace: stack);
>      bool hasCachedMosque = prefs.getBool(MosqueManagerConstant.hasCachedMosque) ?? false;
>      if (!hasCachedMosque) {
>        mosque = null;
>        notifyListeners();
>      }
>      throw e;
>    }
>    ```
> 
> In the scenario you described:
> 
> 1. The user initially requests a mosque that is found. The `hasCachedMosque` flag is set to `true`.
> 2. Later, if that mosque becomes not found (e.g., deleted from the server), the `onItemError` function will be called.
> 3. Because `hasCachedMosque` is `true`, the error handling won't set `mosque` to null or notify listeners.
> 4. The error is still thrown (`throw e;` at the end of `onItemError`).
> 
> This implementation will indeed cause an issue. The error will be thrown, but the UI won't be updated to reflect that the mosque is no longer available. The app will continue to display the cached data of a mosque that no longer exists.

Fixed by last commits @YassinNouh21
